tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/matchstmt/MatchStmtMappingMatchPatternNegativeTest.java

```diff
@@ -117,6 +117,26 @@ public void testMappingMatchPatternTypeNegative() {
         Assert.assertEquals(negativeResult.getErrorCount(), i);
     }
 
+    @Test
+    public void testMappingMatchPatternNegativeSyntax() {
+        CompileResult negativeResult =
+                BCompileUtil.compile("test-src/statements/matchstmt/mapping_match_pattern_syntax_negative.bal");
+        int i = 0;
+        BAssertUtil.validateError(negativeResult, i++,  "missing field match pattern member", 18, 17);
+        BAssertUtil.validateError(negativeResult, i++,
+                "variable '$missingNode$_1' should be declared as constant", 18, 17);
```

Comment:
Tho this is a known issue, can't we just not log that error since it is not needed?

Reply:
How can we not log the error? Special casing here and there is not a good idea.

Reply:
I mean in general

Reply:
Approved since this is not related to this PR